webui/react/src/utils/job.ts

```diff
@@ -43,15 +43,64 @@ export const orderedSchedulers = new Set(
   [ Api.V1SchedulerType.PRIORITY, Api.V1SchedulerType.KUBERNETES ],
 );
 
-export const moveJobToPositionUpdate = (jobId: string, position: number): Api.V1QueueControl => {
+export const moveJobToPositionUpdate = (
+  jobs: Job[],
+  jobId: string,
+  position: number,
+): Api.V1QueueControl | undefined => {
```

Comment:
**question:** Idk a good answer to this, but this might come down to a design decision. I'm not sure if our core utility functions should be able to throw errors, especially when they are used in the React settings. The risk is that the users use them outside of a try/catch block and causes an uncaught error, but they may not encounter that because they didn't force the situation to cause the runtime error.

Unfortunately, Typescript currently doesn't have a type to indicate that the function can throw errors.

wdy think about using comments above functions to indicate that the function can throw errors and hopefully the dev can see the function hinting to see that it can throw errors. https://stackoverflow.com/a/66249151/5402432 see example below:
![Screen Shot 2022-01-31 at 11 09 57 AM](https://user-images.githubusercontent.com/220971/151849086-cb2272f4-b451-466c-b415-6f94c2f72610.png)

Reply:
we gotta check for invalid inputs, it'd be good if we could prevent users from passing invalid values in.. maybe if we exposed a separate validator? but then how do we guarantee that the utility func isn't used without it.

we could always do it like how c or go does which is to return either a return code or a `(value, error)` return type. where the user needs to check if the error is set or not. that can be checked and enforced by typescript. 
as an alternative, the comments sound good.

Reply:
It doesn't show up in my ide when I add it like this

```
diff --git a/webui/react/src/utils/job.ts b/webui/react/src/utils/job.ts
index 9a9bc5569..adb143ecd 100644
--- a/webui/react/src/utils/job.ts
+++ b/webui/react/src/utils/job.ts
@@ -43,6 +43,10 @@ export const orderedSchedulers = new Set(
   [ Api.V1SchedulerType.PRIORITY, Api.V1SchedulerType.KUBERNETES ],
 );
 
+/*
+Create the update request based on a given position for a job.
+@throws {DetError}
+*/
 export const moveJobToPositionUpdate = (
   jobs: Job[],
   jobId: string,

```

ah I wasn't following the correct format. 
https://jsdoc.app/tags-throws.html

I bet there is an eslint rule for that

Reply:
> we could always do it like how `C` or `Go` does which is to return either a return code or a `(value, error)` return type. where the user needs to check if the error is set or not. that can be checked and enforced by typescript. as an alternative, the comments sound good.
>
This sounds like a pretty promising approach `(value, error)` tuple.